### ISSUE TITLE
Struct field alias

### ISSUE THREAD

**Issue by donderom** [markehammons/slinc#215]
Is there any way to map the following definition

```scala
case class foo(type: Int, value: Int) derives Struct
```

where `type` is a keyword in Scala?

**Comment by azolotko** [comment]
@donderom Have you tried
```scala
case class foo(`type`: Int, value: Int) derives Struct
```
?

**Comment by donderom** [comment]
It might work but I can only pass such a structure, and it looks like, in this case, the field name doesn't even matter.

**Comment by markehammons** [comment]
Field names don't matter for struct defiinitions, only the order of members.

**Comment by donderom** [comment]
Ok, so it's only function names that matter, isn't it?

**Comment by markehammons** [comment]
Yes. Function names are used for symbol lookup. There are ways to do platform specific overrides of symbol lookup though.
